### PR TITLE
Add user-friendly logp and logcdf methods

### PR DESCRIPTION
**Pull request by ricardoV94**
This PR adds a `logp` helper similar to the `logcdf` that already existed to facilitate the extraction of `logp` and `logcdf` expressions, similar to how the `.logp` and `.logcdf` methods worked in `v3`:

```python
import pymc3 as pm

# v3:
logp_norm = pm.Normal.dist(0, 1).logp(5.0)
locdf_norm = pm.Normal.dist(0, 1).logcdf(5.0)

# v4:
logp_norm = pm.logp(pm.Normal.dist(0, 1), 5.0)
logcdf_norm = pm.logcdf(pm.Normal.dist(0, 1), 5.0)
```

The file `distributions/logp.py` was renamed to `distributions/logprob.py` to avoid the the name conflict.

### CODE REVIEW

File: pymc3/distributions/logprob.py [hunk 1]
```diff
@@ -342,9 +342,24 @@ def subtensor_logp(op, var, rvs_to_values, indexed_rv_var, *indices, **kwargs):
     return logp_var
 
 
-def logcdf(*args, **kwargs):
+def logp(var, rv_values, **kwargs):
+    """Create a log-probability graph."""
+
+    # Attach the value_var to the tag of var when it does not have one
+    if not hasattr(var.tag, "value_var"):
```
Comment (ricardoV94) [inline pymc3/distributions/logprob.py:349]: This is needed to avoid a warning issued in `aesaraf::rvs_to_value_vars` function.

In main:

```python
import pymc3 as pm
x = pm.Normal.dist(0, 1, size=2)
pm.logcdf(x, [0, 1]).eval()
```
```python
UserWarning: No value variable found for normal_rv{0, (0, 0), floatX, False}.out; the random variable will not be replaced.
```

File: pymc3/tests/test_distributions_random.py [hunk 2]
```diff
@@ -1472,12 +1472,6 @@ def ref_rand(size, mu, sigma, upper):
             pm.TruncatedNormal, {"mu": R, "sigma": Rplusbig, "upper": Rplusbig}, ref_rand=ref_rand
         )
 
-    def test_skew_normal(self):
-        def ref_rand(size, alpha, mu, sigma):
-            return st.skewnorm.rvs(size=size, a=alpha, loc=mu, scale=sigma)
-
-        pymc3_random(pm.SkewNormal, {"mu": R, "sigma": Rplus, "alpha": R}, ref_rand=ref_rand)
-
```
Comment (michaelosthege) [inline pymc3/tests/test_distributions_random.py:1668]: why did you remove it?

**Review by michaelosthege — APPROVED** [review]
I like it!

That reduces one breaking `v3`→`v4` change to something less painful:
```python
my_rv = pm.Normal.dist(0, 2)

# on v3:
my_rv.logp(1.5).eval()

# on v4.
pm.logp(my_rv, 1.5).eval()
```

File: pymc3/tests/test_distributions_random.py (hunk 2 above)
Comment (ricardoV94) [reply pymc3/tests/test_distributions_random.py:1668]: It's one of those old random tests we have been replacing with the BaseTestDistribution class. Should have been removed when we refactored it but I must have missed it.

Also, I checked that it was passing locally before removing it. It did fail in this PR but it's not surprising since it is a non-seeded statistical test

**Review by brandonwillard — COMMENTED** [review]
Generally speaking, don't do too much with respect to the likelihood calculations in PyMC, because we're ultimately going to set PyMC up to use `aeppl`.  Since this looks more like a PyMC-specific interface change, that might not be an issue, though.

**Comment by ricardoV94** [comment]
> Generally speaking, don't do too much with respect to the likelihood calculations in PyMC, because we're ultimately going to set PyMC up to use `aeppl`.  Since this looks more like a PyMC-specific interface change, that might not be an issue, though.

Yeah, this one is actually mimicking a bit what we have in aeppl just to extract a logp/logcdf of a vanilla RV

File: pymc3/tests/test_distributions.py [hunk 3]
```diff
@@ -2398,13 +2389,10 @@ def test_batch_dirichlet_multinomial(self):
         dist = DirichletMultinomial.dist(n=n, a=a)
 
         # Logp should be approx -9.98004998e-06
-        value = at.tensor3(dtype="int32")
-        value.tag.test_value = np.zeros_like(vals, dtype="int32")
-        logp = logpt(dist, value)
-        f = aesara.function(inputs=[value], outputs=logp)
-        expected_logp = np.full(shape=f(vals).shape, fill_value=-9.98004998e-06)
+        dist_logp = logp(dist, vals).eval()
+        expected_logp = np.full_like(dist_lopg, fill_value=-9.98004998e-06)
```
Comment (michaelosthege) [inline pymc3/tests/test_distributions.py]:
```suggestion
        expected_logp = np.full_like(dist_logp, fill_value=-9.98004998e-06)
```

**Review by michaelosthege — COMMENTED** [review]
bad luck

File: pymc3/tests/test_distributions.py (hunk 3 above)
Comment (ricardoV94) [reply pymc3/tests/test_distributions.py]: Yah :)

**Review by michaelosthege — APPROVED** [review]
